src/magic_spell_effect.cpp

```diff
@@ -503,3 +509,245 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
 {
     tp_list.translocate( spell_effect_area( sp, target, spell_effect_blast, caster, true ) );
 }
+
+// hardcoded artifact actives
+void spell_effect::storm( const spell &sp, const tripoint &target )
+{
+    sounds::sound( target, 10, sounds::sound_t::combat, _( "Ka-BOOM!" ), true, "environment",
+                   "thunder_near" );
+    int num_bolts = rng( 2, 4 );
+    for( int j = 0; j < num_bolts; j++ ) {
+        int xdir = 0;
+        int ydir = 0;
+        while( xdir == 0 && ydir == 0 ) {
+            xdir = rng( -1, 1 );
+            ydir = rng( -1, 1 );
+        }
+        int dist = rng( 4, 12 );
+        int boltx = target.x, bolty = target.y;
+        for( int n = 0; n < dist; n++ ) {
+            boltx += xdir;
+            bolty += ydir;
+            sp.create_field( {boltx, bolty, target.z} );
+            if( one_in( 4 ) ) {
+                if( xdir == 0 ) {
+                    xdir = rng( 0, 1 ) * 2 - 1;
+                } else {
+                    xdir = 0;
+                }
+            }
+            if( one_in( 4 ) ) {
+                if( ydir == 0 ) {
+                    ydir = rng( 0, 1 ) * 2 - 1;
+                } else {
+                    ydir = 0;
+                }
+            }
+        }
+    }
+}
+
+void spell_effect::fire_ball( const tripoint &target )
+{
+    explosion_handler::explosion( target, 180, 0.5, true );
+}
+
+void spell_effect::map( const spell &sp, const tripoint &target )
+{
+    player *p = g->critter_at<player>( target );
+    const bool new_map = overmap_buffer.reveal(
+                             point( target.x, target.y ), sp.aoe(), target.z );
+    if( new_map ) {
+        p->add_msg_if_player( m_warning, _( "You have a vision of the surrounding area..." ) );
```

Comment:
Please check that `p` is not null. At the very least add a `assert(p);` somwehere before using `p`.